FinSearchUnified/Tests/Components/ProductStream/RepositoryTest.php

```diff
@@ -47,10 +50,64 @@ public function shopSearchSwitchProvider()
     /**
      * @dataProvider shopSearchSwitchProvider
      *
+     * @param bool $isActive
+     * @param string $shopKey
+     * @param bool $isActiveForCategory
+     * @param bool $checkIntegration
+     * @param bool $isSearchPage
+     * @param bool $isCategoryPage
      * @param bool $prepareCriteria
+     *
+     * @throws \Enlight_Exception
      */
-    public function testUsesOriginalOrDecoratedImplementation($prepareCriteria)
-    {
+    public function testUsesOriginalOrDecoratedImplementation(
+        $isActive,
+        $shopKey,
+        $isActiveForCategory,
+        $checkIntegration,
+        $isSearchPage,
+        $isCategoryPage,
+        $prepareCriteria
+    ) {
+        $request = new \Enlight_Controller_Request_RequestHttp();
+        if ($isCategoryPage) {
+            $request->setParam('sCategory', 3);
```

Comment:
@ihanli in one of the other stories you mentioned that in order for the search to know its a category page, it checks for the sCategory parameter in the request. So that is why I kept it in this condition if its a category page so it renders the correct behaviour